From 4232a69fb7859dd5182ccc22b369b4ff374e5c67 Mon Sep 17 00:00:00 2001
From: Nico <nicorg2529@gmail.com>
Date: Thu, 13 Dec 2018 16:49:36 +0100
Subject: [PATCH 6/9] common: Improve label of I/O sched tuning nodes

* Change the label pattern to reduce pattern matching need.
* Attempt to cover as many paths as possible without using
  /sys/devices/.* which search all subdirs.
* For tuning nodes that is not covered by this, please label
  them in corresponding device trees.

Change-Id: Ibe3efb9da980c165647086163fdb8d41742becca
---
 common/private/file_contexts | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/common/private/file_contexts b/common/private/file_contexts
index 0e85dc1..55b64ba 100644
--- a/common/private/file_contexts
+++ b/common/private/file_contexts
@@ -6,7 +6,10 @@
 /system/bin/mkfs\.ntfs                  u:object_r:mkfs_exec:s0
 
 # I/O Scheduler
-/sys/devices/.*/block/(dm-|mmcblk|sd)[a-z0-9]+/queue(/.*)? u:object_r:sysfs_io_sched_tuneable:s0
+/sys/devices(/platform)?/msm_sdcc\.[0-9]/mmc_host/mmc[0-9]/mmc[0-9]:[a-f0-9]+/block/mmcblk[0-9]/queue(/.*)? u:object_r:sysfs_io_sched_tuneable:s0
+/sys/devices(/platform)?/soc(\.[0-9])?/[a-f0-9]+\.sdhci/mmc_host/mmc[0-9]/mmc[0-9]:[a-f0-9]+/block/mmcblk[0-9]/queue(/.*)? u:object_r:sysfs_io_sched_tuneable:s0
+/sys/devices(/platform)?/soc(\.[0-9])?/[a-f0-9]+\.ufshc/host[0-9]/target[0-9]+:[0-9]+:[0-9]+/[0-9]+:[0-9]+:[0-9]+:[0-9]+/block/sd[a-z]+/queue(/.*)? u:object_r:sysfs_io_sched_tuneable:s0
+/sys/devices/virtual/block/dm-[a-z0-9]+/queue(/.*)? u:object_r:sysfs_io_sched_tuneable:s0
 
 # OTA packages
 /data/lineageos_updates(/.*)?           u:object_r:ota_package_file:s0
-- 
2.17.1

